### PR TITLE
Selector spreading - improving code readability.

### LINKED ISSUE #42577: [minor] Separate getNodeClassificationByLabels so it can be tested.
**Kubernetes version** (use `kubectl version`):

master

**What happened**:

The `getNodeClassificationByLabels` functionality is hidden in a large function, and not explicit, testable.

**What you expected to happen**:

The logic of `CalculateAntiAffinityPriority` should read more clearly, with tests for individual filters.

### PR DESCRIPTION
**Pull request by ravisantoshgudimetla**
**What this PR does / why we need it**:
To improve code readability in selector spreading.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #42577

```release-note
```

### CODE REVIEW

**Comment by ravisantoshgudimetla** [comment]
I signed it!

**Comment by ravisantoshgudimetla** [comment]
cc @jayunit100

File: plugin/pkg/scheduler/algorithm/priorities/selector_spreading.go [hunk 1]
```diff
@@ -60,26 +60,26 @@ func NewSelectorSpreadPriority(
 }
 
 // Returns selectors of services, RCs and RSs matching the given pod.
-func getSelectors(pod *v1.Pod, sl algorithm.ServiceLister, cl algorithm.ControllerLister, rsl algorithm.ReplicaSetLister, ssl algorithm.StatefulSetLister) []labels.Selector {
-	selectors := make([]labels.Selector, 0, 3)
-	if services, err := sl.GetPodServices(pod); err == nil {
+func (s *SelectorSpread) getSelectors(pod *v1.Pod) []labels.Selector {
```
Comment (jayunit100) [inline plugin/pkg/scheduler/algorithm/priorities/selector_spreading.go]: remove this change, it doesnt necessarily improve anything .

File: plugin/pkg/scheduler/algorithm/priorities/selector_spreading.go [hunk 2]
```diff
@@ -206,6 +202,34 @@ func NewServiceAntiAffinityPriority(podLister algorithm.PodLister, serviceLister
 	return antiAffinity.CalculateAntiAffinityPriority
 }
 
+// Classifies nodes into ones with labels and without labels.
+func(s *ServiceAntiAffinity) getNodeClassificationByLabels(nodes []*v1.Node) (map[string]string, []string){
```
Comment (jayunit100) [inline plugin/pkg/scheduler/algorithm/priorities/selector_spreading.go]: thanks, yeah, this addresses the original issue filed, it needs a unit test.

File: plugin/pkg/scheduler/algorithm/priorities/selector_spreading.go (hunk 1 above)
Comment (jayunit100) [inline plugin/pkg/scheduler/algorithm/priorities/selector_spreading.go]: (that said, i understand why you're doing this - the one line call to getSelectors is kinda silly)

File: plugin/pkg/scheduler/algorithm/priorities/selector_spreading.go [hunk 3]
```diff
@@ -60,26 +60,26 @@ func NewSelectorSpreadPriority(
 }
 
 // Returns selectors of services, RCs and RSs matching the given pod.
-func getSelectors(pod *v1.Pod, sl algorithm.ServiceLister, cl algorithm.ControllerLister, rsl algorithm.ReplicaSetLister, ssl algorithm.StatefulSetLister) []labels.Selector {
-	selectors := make([]labels.Selector, 0, 3)
-	if services, err := sl.GetPodServices(pod); err == nil {
+func (s *SelectorSpread) getSelectors(pod *v1.Pod) []labels.Selector {
+	var selectors []labels.Selector
```
Comment (jayunit100) [inline plugin/pkg/scheduler/algorithm/priorities/selector_spreading.go]: i agree with this change -- i think its ok to add this change into the original getSelectors function once you put the logic back (into a function).   i dont see why the `0,3` was declared, but i could be missing something.

**Comment by jayunit100** [comment]
@k8s-bot test this please

**Comment by jayunit100** [comment]
needs gofmt

**Comment by jayunit100** [comment]
yeah extremely low prio , not a bug fix at all or anything like that.

**Comment by jayunit100** [comment]
needs gofmt

File: plugin/pkg/scheduler/algorithm/priorities/selector_spreading.go (hunk 1 above)
Comment (ravisantoshgudimetla) [inline plugin/pkg/scheduler/algorithm/priorities/selector_spreading.go]: The other thing is, extensibility and readability, everytime when someone makes changes, the arguments to this function increase which is weird as we can directly get from struct.

File: plugin/pkg/scheduler/algorithm/priorities/selector_spreading.go (hunk 2 above)
Comment (ravisantoshgudimetla) [inline plugin/pkg/scheduler/algorithm/priorities/selector_spreading.go]: Sure, thats the main idea but I wanted to check, if there is something I missed.

**Comment by ravisantoshgudimetla** [comment]
@timothysc @jayunit100 Yes, the priority is very low. For gofmt I am not seeing any changes for this file, which is weird..

**Comment by jayunit100** [comment]
@ravisantoshgudimetla lets get gofmt fixed on this

**Comment by jayunit100** [comment]
@k8s-bot test this please.

File: plugin/pkg/scheduler/algorithm/priorities/selector_spreading.go [hunk 4]
```diff
@@ -61,7 +61,7 @@ func NewSelectorSpreadPriority(
 
 // Returns selectors of services, RCs and RSs matching the given pod.
 func getSelectors(pod *v1.Pod, sl algorithm.ServiceLister, cl algorithm.ControllerLister, rsl algorithm.ReplicaSetLister, ssl algorithm.StatefulSetLister) []labels.Selector {
-	selectors := make([]labels.Selector, 0, 3)
```
Comment (jayunit100) [inline plugin/pkg/scheduler/algorithm/priorities/selector_spreading.go]: quick question here : why was 3 chosen originally , and is making the slice potentially > 3 possibly going to lead to reduced test coverage?  i assume `no` but just double checking :).

Comment (ravisantoshgudimetla) [inline plugin/pkg/scheduler/algorithm/priorities/selector_spreading.go]: I checked with others as well, it seems its a old change. AFAIK, no its not going to reduced test coverage. Its kind of confusing why we have 3 in first place.(I think previously there were 3 selectors until stateful set was added recently as part of #41708).

**Comment by jayunit100** [comment]
@k8s-bot bazel test this

**Comment by jayunit100** [comment]
@k8s-bot bazel test this

**Comment by jayunit100** [comment]
@k8s-bot gce etcd3 e2e test this

**Comment by ravisantoshgudimetla** [comment]
@jayunit100 @timothysc Any other comments?

**Comment by jayunit100** [comment]
@k8s-bot cvm gce e2e test this

**Comment by jayunit100** [comment]
ping @davidopp @wojtek-t can someone approve this :) thanks

File: plugin/pkg/scheduler/algorithm/priorities/selector_spreading.go [hunk 5]
```diff
@@ -206,6 +206,21 @@ func NewServiceAntiAffinityPriority(podLister algorithm.PodLister, serviceLister
 	return antiAffinity.CalculateAntiAffinityPriority
 }
 
+// Classifies nodes into ones with labels and without labels.
+func (s *ServiceAntiAffinity) getNodeClassificationByLabels(nodes []*v1.Node) (map[string]string, []string) {
+	labeledNodes := map[string]string{}
+	nonLabeledNodes := []string{}
+	for _, node := range nodes {
```
Comment (timothysc) [inline plugin/pkg/scheduler/algorithm/priorities/selector_spreading.go]: fwiw as a follow on... I think we need to start creating cache indexes, b/c we're doing a bunch of operations sub-optimally.

**Comment by davidopp** [comment]
@kubernetes/sig-scheduling-pr-reviews

File: plugin/pkg/scheduler/algorithm/priorities/selector_spreading.go [hunk 6]
```diff
@@ -206,6 +206,21 @@ func NewServiceAntiAffinityPriority(podLister algorithm.PodLister, serviceLister
 	return antiAffinity.CalculateAntiAffinityPriority
 }
 
+// Classifies nodes into ones with labels and without labels.
```
Comment (k82cn) [inline plugin/pkg/scheduler/algorithm/priorities/selector_spreading.go]: s/Classifies/getNodeClassificationByLabels classifies/

File: plugin/pkg/scheduler/algorithm/priorities/selector_spreading.go (hunk 5 above)
Comment (k82cn) [inline plugin/pkg/scheduler/algorithm/priorities/selector_spreading.go]: +1, but node maybe changed during the scheduling cycle, maybe we can share `schedulercache` with predicates & priorities.

File: plugin/pkg/scheduler/algorithm/priorities/selector_spreading.go (hunk 6 above)
Comment (k82cn) [inline plugin/pkg/scheduler/algorithm/priorities/selector_spreading.go]: similar comments to other funcs to make golint happy.

**Pull request merged**